middleware/middleware.js

```diff
@@ -38,36 +42,58 @@ module.exports.fetchRandomGameData = async (req, res, next) => {
         console.log(error.response.status);
       }
       if (retries === maxRetries) {
-        req.session.error = error.response.status;
-        res.redirect(ROUTES.error);
+        return error.response;
       }
     }
     retries++;
   }
-  if (response) {
-    req.gameData = response.data;
+  return response;
+};
+
+module.exports.fetchRandomGameData = async (req, res, next) => {
+  const maxPage = 6720;
 
-    next();
+  const response = await this.fetchFromApi(maxPage);
+  if (response.status >= 400) {
+    req.session.error = response.status;
+    res.redirect(ROUTES.error);
+  }
+  if (response) {
+    req.gameData = response.data.results[0];
   }
+
+  next();
 };
 
-// module.exports.fetchRandomGameDataArr = async (req, res, next) => {
-//   const pageNumArr = generateUniqueRandomArr();
-//   let gameDataArr = [];
-//   for (let num of pageNumArr) {
-//     let response = await axios.get(
-//       `${apiUrl}games?key=${process.env.RAWG_KEY}&page=${pageNumArr[num]}&${apiParams}`
-//     );
-//     response.data.results.forEach((result) => {
-//       if (result.background_image) {
-//         gameDataArr.push({ name: result.name, image: result.background_image });
-//       }
-//     });
-//   }
-//   console.log(gameDataArr);
-//   req.gameDataArr = gameDataArr;
-//   next();
-// };
+module.exports.fetchRandomGameDataArr = async (req, res) => {
+  const maxPage = 168;
+  const pageSize = 40;
+  const gameNumArr = generateUniqueRandomArr(40, 4);
+
+  let filteredGamesData = { gamesArray: [] };
+
+  const response = await this.fetchFromApi(maxPage, pageSize);
```

Comment:
Interesting usage of `this` keyword. Why not just use `fetchFromApi`?

Reply:
It was something that the VS Code added automatically and without it `fetchFromApi` didnt work, so i left it there

Reply:
As long as it works I guess. I would have done it differently for example, all you would need to do here is declare `fetchFromApi` as a constant:

```
const fetchFromApi = async (maxPage, pageSize = 1) => {
  let response = null;
  let success = false;
  let retries = 1;
  const maxRetries = 5;

  while (retries <= maxRetries && !success) {
    try {
      const randGameId = generateRandomNum(maxPage);
      response = await axios.get(
        `${apiUrl}games?key=${process.env.RAWG_KEY}&page=${randGameId}&page_size=${pageSize}&${apiParams}`
      );
      if (response.data.results[0].background_image) {
        success = true;
        break;
      }
    } catch (error) {
      if (process.env.IN_DEVELOPMENT === "true") {
        console.log(error.response.status);
      }
      if (retries === maxRetries) {
        return error.response;
      }
    }
    retries++;
  }
  return response;
};
```

Then export it in the same way you did before: 

```
module.exports.fetchFromApi = fetchFromApi;
```

Reply:
Ow, i see. I simply did not know that I could do it this way and it would work. But i guess there isn't much of a reason to change it since it seems to work just fine as is, unless there are issues that might potentially arise with the use of `this`.